pkg/controller/elasticsearch/client/test_fixtures/autoscaling.go

```diff
@@ -0,0 +1,140 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```

Comment:
Can't we put this and all the tests under the special `client_test` package because it is only needed during tests? Alternatively, it can be stored as a JSON file in the `testdata` directory and loaded at runtime?

Reply:
This file belongs to the `fixtures` package, maybe I'm missing something but I can't find the `client_test` package.

I didn't want to introduce a `testdata` directory here only to be consistent with what we do [for other tests](https://github.com/elastic/cloud-on-k8s/blob/633982dd077c1d41519e9aa6995159615c06f92c/pkg/controller/elasticsearch/client/test_fixtures/nodes.go). But I agree that some of those tests would deserve it.

Reply:
It's a Go testing idiom that we don't use in the ECK code base much. You can change the package name of `*_test.go` files to `<package>_test` and that's legal. So, what I am suggesting is that instead of using `package client` in the test files, we can switch to `package client_test` to keep the test code separate.

We use `testdata` in several places as well: https://github.com/elastic/cloud-on-k8s/tree/master/pkg/controller/common/certificates/testdata